src/updater.cpp

```diff
@@ -139,48 +139,44 @@ namespace appimage {
 
                     updateInformation = rawUpdateInformation.data();
                 } else if (appImageType == 2) {
-                    // check whether update information can be found inside the file by calling objdump
-
-                    // first of all, check whether there is an objdump binary next to the current one (probably
-                    // the bundled one in the AppImages of AppImageUpdate), otherwise use the system wide
-                    std::string objdump;
-
-                    {
-                        // TODO: replace this Linux specific solution with something platform independent
-                        std::vector<char> buffer(4096);
-                        readlink("/proc/self/exe", buffer.data(), buffer.size());
-
-                        auto pathToBinary = std::string(buffer.data());
-                        auto slashPos = pathToBinary.find_last_of('/');
-                        auto bundledObjdump = pathToBinary.substr(0, slashPos) + "/objdump";
-
-                        if (isFile(bundledObjdump)) {
-                            objdump = bundledObjdump;
-                        } else {
-                            objdump = "objdump";
+                   {
+                     unsigned offset = 0, length = 0;
+                     uint8_t * data;
+                     Elf64_Ehdr * elf64;
+                     Elf64_Shdr * shdr64;
+                     Elf32_Ehdr * elf32;
+                     Elf32_Shdr * shdr32;
+                     int i, fd = open(pathToAppImage.c_str(), O_RDONLY);
+                     data = (uint8_t * ) mmap(NULL, lseek(fd, 0, SEEK_END), PROT_READ, MAP_SHARED, fd, 0);
+                     elf64 = (Elf64_Ehdr * ) data;
+                     shdr64 = (Elf64_Shdr * )(data + elf64-> e_shoff);
+                     if (elf64-> e_ident[EI_CLASS] != ELFCLASS64) {
+                     elf32 = (Elf32_Ehdr * ) data;
+                     shdr32 = (Elf32_Shdr * )(data + elf32-> e_shoff);
+
+                     char * strTab = (char * )(data + shdr32[elf32-> e_shstrndx].sh_offset);
+                     for (i = 0; i < elf32-> e_shnum; i++) {
+                     if (strcmp( & strTab[shdr32[i].sh_name], ".upd_info") == 0) {
+                        offset = shdr32[i].sh_offset;
+                        length = shdr32[i].sh_size;
+                      }
+                     }
+                     } else {
```

Comment:
Follow-up for comment in line 153: if the block above this line would be properly indented, this was much easier to read.